### PR TITLE
refactor: ADI modules

### PR DESCRIPTION
**Pull request by Tropix126**
## Describe the changes this PR makes. Why should it be merged?
This PR refactors the modules relating to ADI in `vexide_devices`. This is mainly documentation improvements, but also:
- Adds a proper `level` getter to `AdiDigitalOut` that calls to VEXos rather than just statically storing the level and hoping its low on program start.
- Adds a `toggle` method to `AdiDigitalOut`.
Removes `AdiSolenoid`.
- Resolves some rustdoc issues and improves docs in other unrelated modules.

### CODE REVIEW

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]
changelog :3

File: packages/vexide-devices/src/adi/light_sensor.rs [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
+//! The light sensor only measures light in the visible spectrum. It cannot detect
+//! infrared or ultraviolet sources.
+//!
+//! # Effective Range
+//!
+//! Effective range is dependent on both the intensity of the source and the surrounding
+//! enviornment. Darker ambient surroundings with a brighter source will result in a
+//! greater effective range.
+//!
+//! That being said, the sensor generally has a usable range of up to 6 feet, meaning it
```
Comment (Gavin-Niederman) [inline packages/vexide-devices/src/adi/light_sensor.rs:19]: Not really a reason not to merge, but has this been verified?

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]

File: packages/vexide-devices/src/adi/light_sensor.rs (hunk 1 above)
Comment (Tropix126) [reply packages/vexide-devices/src/adi/light_sensor.rs:19]: This comment is based on vex's own technical documentation.

> 	Usable range of 0 to 6 feet

**Review by Gavin-Niederman — APPROVED** [review]
LGTM

**Pull request merged**